### PR TITLE
Basic wheel building support

### PR DESCRIPTION
**Pull request by kgaughan**
These are the code changes mentioned in #54, as well as fixing the dependency list in `setup.py` to match `requirements.txt`.

With this, a wheel can be built like so:
```
$ python3 -m venv .venv
$ .venv/bin/pip install -r dev-requirements.txt
$ .venv/bin/python setup.py sdist bdist_wheel
```

### CODE REVIEW

**Comment by kgaughan** [comment]
I had to remove the pegged wheel version because of 3.4 compatibility. I also added test builds against 3.7 and 3.8.

**Comment by DavidMuller** [comment]
Thanks @kgaughan - I've used this as a base to publish a wheel (mostly contained to PR https://github.com/DavidMuller/aws-requests-auth/pull/56)

File: setup.py [hunk 1]
```diff
@@ -10,7 +10,10 @@
     url='https://github.com/davidmuller/aws-requests-auth',
     description='AWS signature version 4 signing process for the python requests module',
     long_description='See https://github.com/davidmuller/aws-requests-auth for installation and usage instructions.',
-    install_requires=['requests>=0.14.0'],
+    install_requires=[
+        'botocore>=1.7.8',
```
Comment (DavidMuller) [inline setup.py:14]: It's a bit odd, but we don't require `botocore` as a strict dependency of the package:

> [botocore (the core functionality of boto3) is not a strict requirement of aws-requests-auth, but we do provide some convenience methods if you'd like to use botocore to automatically retrieve your AWS credentials for you.](https://github.com/DavidMuller/aws-requests-auth#using-boto-to-automatically-gather-aws-credentials)

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,2 @@
-mock==1.3.0
 requests==2.20.0
```
Comment (DavidMuller) [inline requirements.txt:1]: right now `requirements.txt` is just for people developing new features `aws_requests_auth` - it is not involved in the `setup.py` / package publishing component of the repo. So, https://github.com/DavidMuller/aws-requests-auth/pull/56 ("Prep for wheel support on pypi") excludes this change

**Pull request closed without merging**